docs/build-macos.md

```diff
@@ -0,0 +1,52 @@
+macOS builds can be created using:
+
+- Clang, using Meson buildsystem and Homebrew for installing dependencies
+  (*recommended*)
+- GCC, using Meson buildsystem and Homebrew for installing dependencies.
+- Clang or GCC, using Meson buildsystem and MacPorts for dependencies.
+
+Only builds using Apple's Clang can link with Apple's Core SDK.
```

Comment:
We recommending using Brew and Clang, because for the latter only Apple's Clang can link with Apple's Core SDK.